code/common/os.q

```diff
@@ -14,4 +14,5 @@ df:{(`$("/";"\\")[NT]sv -1_v;`$-1#v:("/";"\\")[NT]vs pth(string x;x)[10h=type x]
 run:{system"q ",x}
 kill:{[p]@[(`::p);"\\\\";1];}
 sleep:{x:string x; system("sleep ",x;"timeout /t ",x," >nul")[NT]}

```

Comment:
Conditionals are used elsewhere yet this is indexed, time to change whilst other things are being edited?
@jonathonmcmurray @jonnypress

Reply:
if I change this I think it would be smart to use just one conditional and then maybe index in a dictionary for the specific command. I.e have a table like ```table:([]command:`remove`list;linux:`rm`ls;windows:`del`dir)``` and then based on the os create a dictionary that I can then index in? I don't think it would look good to use and if-statement for every command. Opinions?

Reply:
The way that seems most natural to me would be to define to dictionaries, `win` and `unix` e.g.

```q
win.sleep:{...}
unix.sleep:{...}
win.ren:{...}
unix.ren:{...}
```

and then based on OS join approriate dict to `.os` namespace (which would already contain any functions that don't depend on OS, e.g. `.os.run`)

```q
.os,:(unix;win)NT
```

or something.

But maybe this leads to a lot of duplicated code in cases where only one command is changing (e.g. `mv` vs `move`)